http/src/request/guild/ban/get_bans.rs

```diff
@@ -7,37 +7,105 @@ use crate::{
 };
 use twilight_model::{
     guild::Ban,
-    id::{marker::GuildMarker, Id},
+    id::{
+        marker::{GuildMarker, UserMarker},
+        Id,
+    },
 };
+use twilight_validate::request::{
+    get_guild_bans_limit as validate_get_guild_bans_limit, ValidationError,
+};
+
+struct GetBansFields {
+    after: Option<Id<UserMarker>>,
+    before: Option<Id<UserMarker>>,
+    limit: Option<u16>,
+}
 
 /// Retrieve the bans for a guild.
 ///
 /// # Examples
 ///
-/// Retrieve the bans for guild `1`:
+/// Retrieve the first 25 bans of a guild after a particular user ID:
 ///
 /// ```no_run
+/// use std::env;
 /// use twilight_http::Client;
 /// use twilight_model::id::Id;
 ///
 /// # #[tokio::main]
 /// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
-/// let client = Client::new("my token".to_owned());
+/// let client = Client::new(env::var("DISCORD_TOKEN")?);
 ///
 /// let guild_id = Id::new(1);
+/// let user_id = Id::new(2);
 ///
-/// let bans = client.bans(guild_id).exec().await?;
+/// let response = client.bans(guild_id).after(user_id).exec().await?;
+/// let bans = response.models().await?;
+///
+/// for ban in bans {
+///     println!("{} was banned for: {:?}", ban.user.name, ban.reason);
+/// }
 /// # Ok(()) }
 /// ```
 #[must_use = "requests must be configured and executed"]
 pub struct GetBans<'a> {
+    fields: GetBansFields,
     guild_id: Id<GuildMarker>,
     http: &'a Client,
 }
 
 impl<'a> GetBans<'a> {
     pub(crate) const fn new(http: &'a Client, guild_id: Id<GuildMarker>) -> Self {
-        Self { guild_id, http }
+        Self {
+            fields: GetBansFields {
+                after: None,
+                before: None,
+                limit: None,
+            },
+            guild_id,
+            http,
+        }
+    }
+
+    /// Set the user ID after which to retrieve bans.
+    ///
+    /// Mutually exclusive with [`before`]. If both are provided then [`before`]
+    /// is respected.
+    ///
+    /// [`before`]: Self::before
+    pub const fn after(mut self, user_id: Id<UserMarker>) -> Self {
+        self.fields.after = Some(user_id);
+
+        self
+    }
+
+    /// Set the user ID before which to retrieve bans.
+    ///
+    /// Mutually exclusive with [`after`]. If both are provided then [`before`]
+    /// is respected.
```

Comment:
Should we note that before takes precedence in the case both are supplied?

Reply:
Is that not already done?

Reply:
Yes, but I somehow lost my ability to read.